src/python/pants/backend/python/util_rules/interpreter_constraints.py

```diff
@@ -259,6 +259,61 @@ def to_poetry_constraint(self) -> str:
             return "*"
         return " || ".join(specifiers)
 
+    def enumerate_python_versions(
+        self, interpreter_universe: Iterable[str]
+    ) -> set[tuple[int, int, int]]:
```

Comment:
Now that this function is graduating to be public, it should probably return `Iterator[tuple[int, int, int]]` and use `yield`.

Reply:
That feels like a step backwards in terms of clarity to me (and will make the call sites slightly more complex)